connection/connection.go

```diff
@@ -203,11 +210,30 @@ func (cmm ExtendedCSIMetricsManager) RecordMetricsClientInterceptor(
 	start := time.Now()
 	err := invoker(ctx, method, req, reply, cc, opts...)
 	duration := time.Since(start)
-	cmm.RecordMetrics(
-		method,   /* operationName */
-		err,      /* operationErr */
-		duration, /* operationDuration */
-	)
+
+	// record migration status
+	additionalInfo := ctx.Value(AdditionalInfoKey)
+	migrated := "false"
+	if additionalInfo != nil {
+		migrated = additionalInfo.(AdditionalInfo).Migrated
+	}
+	cmmv, metricsErr := cmm.WithLabelValues(map[string]string{metrics.LabelMigrated: migrated})
+	if metricsErr != nil {
```

Comment:
What happens if the sidecar didn't define the "migrated" label? This will cause `WithLabelValues` to fail, right?

I think it would be better to let sidecars opt into this feature. This means only calling `WithLabelValues` if there is some non-empty value to be recorded.

Reply:
If the sidecar did not define the migrated label, we will fall back to use the old cmm to record metrics. So the old sidecar metrics will still be the same. See comments on line 229